extensions/sql-assessment/src/assessmentResultGrid.ts

```diff
@@ -29,20 +29,35 @@ export class AssessmentResultGrid implements vscode.Disposable {
 
 	private asmtType!: AssessmentType;
 
-	private readonly checkIdColOrder = 4;
+	private readonly checkIdColOrder = 5;
 	private readonly targetColOrder = 0;
 
 	public get component(): azdata.Component {
 		return this.rootContainer;
 	}
 
-	public constructor(view: azdata.ModelView) {
+	public constructor(view: azdata.ModelView, extensionContext: vscode.ExtensionContext) {
 		const headerCssClass = 'no-borders align-with-header';
 		this.table = view.modelBuilder.table()
 			.withProperties<azdata.TableComponentProperties>({
 				data: [],
 				columns: [
-					{ value: LocalizedStrings.TARGET_COLUMN_NAME, headerCssClass: headerCssClass, width: 125 },
+					{
+						value: LocalizedStrings.TARGET_COLUMN_NAME, headerCssClass: headerCssClass, width: 125,
+						options: {
+							iconProviderColumn: 'targetImage',
+							imageCollection: [{
```

Comment:
Why wouldn't you just have the actual row values themselves contain the path? That seems easier to use than having a collection like this.

Reply:
Row values are string types, I can't set { light: string | vscode.Uri; dark: string | vscode.Uri } object there. But I want to support dark/light theme updates. This functionality covered by createIconCssClass function on the modelComponents utils, but an extension doesn't have access to it.

Reply:
I think it's reasonable to extend the value to accept the icon path type as well. That would be a much cleaner approach I think.

@alanrenmsft any thoughts on this?

Reply:
At the very least if we do this I think I'd rather have the collection be a key/value map instead of using indexes - those are easier to track and debug through.